rmw_zenoh_cpp/src/detail/liveliness_utils.hpp

```diff
@@ -227,6 +230,35 @@ std::optional<rmw_qos_profile_t> keyexpr_to_qos(const std::string & keyexpr);
 ///=============================================================================
 /// Convert a Zenoh id to a string.
 std::string zid_to_str(const z_id_t & id);
+
+///=============================================================================
+// Helper function to convert string to size_t.
+// The function is templated to enable conversion to size_t or std::size_t.
+template<typename T>
+std::optional<T> str_to_size_t(const std::string & str, const T default_value)
```

Comment:
This whole function doesn't need to live in the header file; we can just keep it as an anonymous namespace function within `liveliness_utils.cpp`.